src/fuzz_demangle.cc

```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#include "demangle.h"
+
+using namespace GOOGLE_NAMESPACE;
+
+extern "C" int LLVMFuzzerTestOneInput(const unsigned char *Data, unsigned Size) {
```

Comment:
Why `LLVM` here? Is the naming scheme prescribed by the fuzzing engine? If so, can we use different engines at all as the CMake setup suggests?

Reply:
> Why `LLVM` here? Is the naming scheme prescribed by the fuzzing engine?

The name is indeed prescribed cf https://llvm.org/docs/LibFuzzer.html#id23

>  If so, can we use different engines at all as the CMake setup suggests?

oss-fuzz uses different fuzzing engines : libfuzzer, afl, honggfuzz... It manages to do so with every fuzz target having `LLVMFuzzerTestOneInput` as its entry point. If I am correct, to use other engines, there are simply wrappers that call `LLVMFuzzerTestOneInput`...